plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QRefreshPanelAction.kt

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.amazonq
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.util.messages.Topic
+import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindow
+import software.aws.toolkits.resources.AmazonQBundle
+import software.aws.toolkits.resources.message
+import java.util.EventListener
+
+class QRefreshPanelAction : DumbAwareAction(AmazonQBundle.message("amazonq.refresh.panel"), "abcs", AllIcons.Actions.Refresh) {
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        // recreate chat browser
+        AmazonQToolWindow.getInstance(project).disposeAndRecreate()
+        // recreate signin browser
+        QWebviewPanel.getInstance(project).disposeAndRecreate()
+        RefreshQChatPanelButtonPressedListener.notifyRefresh()
```

Comment:
listeners are not the right pattern for a single sender -> single receiver
let's fix this later